src/OpenTK/Platform/SDL2/Sdl2DisplayDeviceDriver.cs

```diff
@@ -86,17 +90,56 @@ private Rectangle TranslateBounds(Rect rect)
             return new Rectangle(rect.X, rect.Y, rect.Width, rect.Height);
         }
 
-        public override bool TryChangeResolution(DisplayDevice device, DisplayResolution resolution)
+        public bool TryChangeResolution(object device, DisplayResolution resolution)
         {
             Sdl2Factory.UseFullscreenDesktop = false;
             return true;
         }
 
-        public override bool TryRestoreResolution(DisplayDevice device)
+        public bool TryRestoreResolution(object device)
         {
             Sdl2Factory.UseFullscreenDesktop = true;
             return true;
         }
+
+        public  DisplayResolution GetResolution(object device)
+        {
+            var sdlDevice = (SdlDevice)device;
+
+            Rect bounds;
+            SDL.GetDisplayBounds(sdlDevice.Index, out bounds);
+
+            DisplayMode currentMode;
+            SDL.GetCurrentDisplayMode(sdlDevice.Index, out currentMode);
+
+            return new DisplayResolution(
+                bounds.X, bounds.Y,
+                currentMode.Width, currentMode.Height,
+                TranslateFormat(currentMode.Format),
+                currentMode.RefreshRate);
+        }
+
+        public bool GetIsPrimary(object device)
+        {
+            var sdlDevice = (SdlDevice)device;
+            return sdlDevice.Index == 0;
+        }
+
+        public IList<DisplayResolution> GetAvailableResolutions(object device)
```

Comment:
That is small API change, everything else in this PR is internal only. Changing to IReadOnlyList should work ok though.